lib/roby/exceptions.rb

```diff
@@ -219,83 +219,90 @@ def handle_exception(exception_object)
+        backtrace_bottom = []
+        while !original_backtrace.empty? &&
+              filter_out.none? { |rx| rx.match?(original_backtrace.last) }
+            backtrace_bottom.unshift original_backtrace.pop
+        end
+
+        got_user_line = false
+        backtrace = original_backtrace.enum_for(:each_with_index).map do |line, idx|
+            case line
+            when /in `poll_handler'$/
+                got_user_line = true
+                line.gsub(/:in.*/, ":in the polling handler")
+            when /in `event_command_(\w+)'$/
+                got_user_line = true
+                line.gsub(/:in.*/, ":in command for '#{$1}'")
+            when /in `event_handler_(\w+)_(?:[a-f0-9]+)'$/
+                got_user_line = true
+                line.gsub(/:in.*/, ":in event handler for '#{$1}'")
+            else
+                if original_backtrace.size > idx + 4 &&
+                   original_backtrace[idx + 1] =~ /in `call'$/ &&
+                   original_backtrace[idx + 2] =~ /in `call_handlers'$/ &&
+                   original_backtrace[idx + 3] =~ /`each'$/ &&
+                   original_backtrace[idx + 4] =~ /`each_handler'$/
 
-            got_user_line = false
-            backtrace = original_backtrace.enum_for(:each_with_index).map do |line, idx|
-                case line
-                when /in `poll_handler'$/
                     got_user_line = true
-                    line.gsub(/:in.*/, ":in the polling handler")
-                when /in `event_command_(\w+)'$/
-                    got_user_line = true
-                    line.gsub(/:in.*/, ":in command for '#{$1}'")
-                when /in `event_handler_(\w+)_(?:[a-f0-9]+)'$/
-                    got_user_line = true
-                    line.gsub(/:in.*/, ":in event handler for '#{$1}'")
+                    line.gsub(/:in /, ":in event handler, ")
                 else
-                    if original_backtrace.size > idx + 4 &&
-                       original_backtrace[idx + 1] =~ /in `call'$/ &&
-                       original_backtrace[idx + 2] =~ /in `call_handlers'$/ &&
-                       original_backtrace[idx + 3] =~ /`each'$/ &&
-                       original_backtrace[idx + 4] =~ /`each_handler'$/
-
-                        got_user_line = true
-                        line.gsub(/:in /, ":in event handler, ")
-                    else
-                        is_user = filter_out.none? { |rx| rx.match?(line) }
-                        got_user_line ||= is_user
-                        if !got_user_line || is_user
-                            case line
-                            when /^\(eval\):\d+:in `each(?:_handler)?'/
-                                nil
-                            else
-                                line
-                            end
+                    is_user = filter_out.none? { |rx| rx.match?(line) }
+                    got_user_line ||= is_user
+                    if !got_user_line || is_user
```

Comment:
got_user_line is always true when is_user is, so this could become `unless got_user_line`

Reply:
This is true ... but it means that when is_user is T, the existing condition is T while your proposed condition is F

| `got_user_line` | `is_user` | `!got_user_line or is_user` | `!got_user_line` |
| --- | --- | --- | --- |
| F | F| T | T |
| F | T| - | - |
| T | F| F | F |
| T | T| T | F |